statsmodels/stats/contingency_tables.py

```diff
@@ -0,0 +1,1215 @@
+        # etc.
+        self._apb = table[0, 0, :] + table[0, 1, :]
+        self._apc = table[0, 0, :] + table[1, 0, :]
+        self._bpd = table[0, 1, :] + table[1, 1, :]
+        self._cpd = table[1, 0, :] + table[1, 1, :]
+        self._ad = table[0, 0, :] * table[1, 1, :]
+        self._bc = table[0, 1, :] * table[1, 0, :]
+        self._apd = table[0, 0, :] + table[1, 1, :]
+        self._dma = table[1, 1, :] - table[0, 0, :]
+        self._n = table.sum(0).sum(0)
+
+
+    @classmethod
+    def from_data(cls, var1, var2, strata, data):
+        """
+        Construct a StratifiedTables object from data.
+
+        Parameters
+        ----------
+        var1 : int or string
+            The column index or name of `data` containing the variable
+            defining the rows of the contingency table.  The variable
+            must have only two distinct values.
+        var2 : int or string
+            The column index or name of `data` containing the variable
+            defining the columns of the contingency table.  The variable
+            must have only two distinct values.
+        strata : int or string
+            The column index of name of `data` containing the variable
+            defining the strata.
+        data : array-like
+            The raw data.  A cross-table for analysis is constructed
+            from the first two columns.
+
+        Returns
+        -------
+        A StratifiedTables instance.
+        """
+
+        if not isinstance(data, pd.DataFrame):
+            data1 = pd.DataFrame(index=data.index, column=[var1, var2, strata])
+            data1.loc[:, var1] = data[:, var1]
+            data1.loc[:, var2] = data[:, var2]
+            data1.loc[:, strata] = data[:, strata]
+        else:
+            data1 = data
+
+        gb = data1.groupby(strata).groups
+        tables = []
+        for g in gb:
+            ii = gb[g]
+            tab = pd.crosstab(data1.loc[ii, var1], data1.loc[ii, var2])
+            tables.append(tab)
+        return cls(tables)
+
+
+    def test_null_odds(self, correction=False):
+        """
+        Test that all tables have odds ratio equal to 1.
+
+        This is the 'Mantel-Haenszel' test.
+
+        Parameters
+        ----------
+        correction : boolean
+            If True, use the continuity correction when calculating the
+            test statistic.
+
+        Returns the chi^2 test statistic and p-value.
+        """
+
+        stat = np.sum(self.table[0, 0, :] - self._apb * self._apc / self._n)
+        stat = np.abs(stat)
+        if correction:
+            stat -= 0.5
+        stat = stat**2
+        denom = self._apb * self._apc * self._bpd * self._cpd
+        denom /= (self._n**2 * (self._n - 1))
+        denom = np.sum(denom)
+        stat /= denom
+
+        # df is always 1
+        pvalue = 1 - stats.chi2.cdf(stat, 1)
+
+        return stat, pvalue
+
+
+    @cache_readonly
+    def common_odds(self):
```

Comment:
I would prefer postfix `pooled`, e.g. oddsratio_pooled (standard terminology for pooled sample)